crates/sargon/src/core/utils/image_url_utils.rs

```diff
@@ -0,0 +1,116 @@
+use crate::prelude::*;
+use crate::types::*;
+
+pub fn is_vector_image(url: &Url, image_type: VectorImageType) -> bool {
+    let url_string = url.to_string();
+    let query_parameters = url
+        .query_pairs()
+        .into_owned()
+        .collect::<HashMap<String, String>>();
+    let image_url_string = query_parameters
+        .get("imageOrigin")
+        .map(|s| s.as_str())
+        .unwrap_or(&url_string);
+
+    image_url_string
+        .starts_with(&format!("data:image/{}", image_type.data_url_type()))
+        || image_url_string
+            .to_lowercase()
+            .ends_with(image_type.url_extension())
+}
+
+pub fn make_image_url(
+    url: &str,
+    image_service_url: &str,
+    width: u32,
+    height: u32,
+) -> Result<Url> {
+    const MIN_SIZE: u32 = 64;
+
+    let parsed_url = parse_url(url)?;
+    let image_origin = url_encode(url);
+    let image_size =
+        format!("{}x{}", width.max(MIN_SIZE), height.max(MIN_SIZE));
+    let mut image_url = format!(
+        "{}/?imageOrigin={}&imageSize={}",
+        image_service_url, image_origin, image_size
```

Comment:
@danvleju-rdx I think we need to also check if the `image_service_url` contains `/` at the end. Since we explicitly add one ourselves.

Or maybe better to use [Url's `set_query`](https://docs.rs/url/latest/url/struct.Url.html#method.set_query) and let it  manage all the corner cases?

Reply:
In android when I need to use this method the following:

```
val url: Url = "https://image-service-dev.extratools.works".toUrl()

println(url.toString()) 
// prints out https://image-service-dev.extratools.works/
// so if I use the method in rust it will result in
// https://image-service-dev.extratools.works//?imageOrigin=https%3A%2F%2Fsvgshare.com%2Fi%2FU7z.svg&imageSize=1024x1024&format=png
// which is not valid
```

Reply:
`set_query` sounds good. Let me try to see if it doesn't break the data URLs

Reply:
I've made the changes [here](https://github.com/radixdlt/sargon/pull/221/commits/93942ef63afea3811c6a0960642e64f2fe4b71b8). LMK if this resolves the issue.